config/configtls/configtls.go

```diff
@@ -75,6 +75,11 @@ type Config struct {
 	ReloadInterval time.Duration `mapstructure:"reload_interval"`
 }
 
+// NewDefaultConfig creates a new TLSSetting with any default values set.
+func NewDefaultConfig() Config {
```

Comment:
Is there a need for this from users perspective? Or we think it is enough only the Client/Server? Happy to keep it if we have few needs.

Reply:
I dont have a specific user cases, only following the pattern defined in https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md#default-configuration. The pattern implies to me that we should have a `NewDefault` for all public config APIs.  It is probably overkill, but it does allow us to add new default values to `Config` in the future if we want.

Reply:
I'd rather have the `NewDefaultConfig` function.

One thing is that we can use it in the `NewDefaultClientConfig` and `NewDefaultServerConfig`, although admittedly it doesn't need to be exported for this.

The other thing is that users can create their own configs that embed the `configtls.Config` struct and then they can use the `configtls.NewDefaultConfig` to create their instances of their configs.

I hope you don't ask me for potential examples of other configs that embed `configtls.Config`, because I don't have any such examples. :smile: